### PR TITLE
[Dropbox] Add Unstructured file decoding to Dropbox

### PR DESCRIPTION
**Pull request by tianjing-li**
### What's being changed:

- Adds unstructured client to Dropbox connector, with async requests
- Adds serialization for Dropbox results, gathers files for batch Unstructured requests

### How did you test this change (include any code snippets, API requests, screenshots, or gifs):

- Tested the content being properly decoded locally, batch async requests also work

### CODE REVIEW

File: dropbox/provider/provider.py [hunk 1]
```diff
@@ -17,12 +24,35 @@ def search(query: str, oauth_token: str = None) -> list[dict[str, Any]]:
...
 
+    # Group for batch Unstructured requests
+    files = [
+        (result["id"], result["title"], result["raw_content"]) for result in results
+    ]
+    unstructured_content = unstructured_client.batch_get(files)
+
+    # Now build text field based off Unstructured return
+    for result in results:
+        del result["raw_content"]
```
[the remark this answers is not in the document]
Comment (tianjing-li) [reply dropbox/provider/provider.py:46]: It's the raw encoded file contents that get decoded by Unstructured, I figured it wouldn't help much to return it, but if we don't mind it in the response then I can remove the `del`

File: dropbox/provider/unstructured.py [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+    def __init__(self, unstructured_base_url, api_key):
+        self.get_content_url = f"{unstructured_base_url}/general/v0/general"
+        self.headers = {"unstructured-api-key": api_key}
+        # Manually cache because functools.lru_cache does not support async methods
+        self.cache = OrderedDict()
+        self.start_session()
+
+    def start_session(self):
+        self.loop = asyncio.new_event_loop()
+        self.session = aiohttp.ClientSession(loop=self.loop)
```
[the remark this answers is not in the document]
Comment (tianjing-li) [reply dropbox/provider/unstructured.py:24]: Done

File: dropbox/provider/unstructured.py [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+    if unstructured is not None:
+        return unstructured
+
+    # Fetch environment variables
+    assert (
+        unstructured_base_url := app.config.get("UNSTRUCTURED_BASE_URL")
+    ), "DROPBOX_UNSTRUCTURED_BASE_URL must be set"
+    assert (
+        api_key := app.config.get("UNSTRUCTURED_API_KEY")
+    ), "DROPBOX_UNSTRUCTURED_API_KEY must be set"
```
[the remark this answers is not in the document]
Comment (tianjing-li) [reply dropbox/provider/unstructured.py:100]: Done

File: dropbox/provider/unstructured.py [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+
+    def cache_get(self, key):
+        self.cache.move_to_end(key)
+
+        return self.cache[key]
+
+    def cache_put(self, key, item):
+        self.cache[key] = item
+
+        if len(self.cache) > CACHE_SIZE:
```
[the remark this answers is not in the document]
Comment (tianjing-li) [reply dropbox/provider/unstructured.py:38]: Good point, sure

[the remark this answers is not in the document]
Comment (tianjing-li) [reply dropbox/provider/unstructured.py]: Done - `getsizeof` on the dictionary itself isn't accurate, I instead summed up the sizes of all the dictionary's values, sent you explanation on Slack

File: dropbox/.env-template [hunk 5]
```diff
@@ -3,4 +3,8 @@ DROPBOX_APP_KEY=
 DROPBOX_APP_SECRET=
```
Comment (EugeneLightsOn) [inline dropbox/.env-template:3]: The variables DROPBOX_APP_KEY and DROPBOX_APP_SECRET are not used. Let's remove them here and add notes in the README related to the connector registration block

**Review by EugeneLightsOn — COMMENTED** [review]
Looks good to me, just one comment

File: dropbox/.env-template (hunk 5 above)
Comment (tianjing-li) [reply dropbox/.env-template]: Done - removed, there is a section in the README that requires the app secret and key

File: dropbox/.env-template [hunk 6]
```diff
@@ -1,6 +1,8 @@
...
-DROPBOX_APP_SECRET=
 DROPBOX_SEARCH_LIMIT=5
 DROPBOX_PATH=
-DROPBOX_CONNECTOR_API_KEY=
\ No newline at end of file
+DROPBOX_CONNECTOR_API_KEY=
+
+# Unstructured
+DROPBOX_UNSTRUCTURED_BASE_URL=
+DROPBOX_UNSTRUCTURED_API_KEY=
```
Comment (daniel-cohere) [inline dropbox/.env-template:8]: just a newline missing i think

**Review by daniel-cohere — APPROVED** [review]

**Pull request merged**